### DISCUSSION TITLE
Do equality constraints ease solution finding or make it harder?

### DISCUSSION THREAD

**Discussion by ogencoglu** [qpsolvers/qpsolvers#283] [General]
This is a bit generic question regarding quadratic programming.

In general, do equality constraints, i.e. `Ax=b`, help the solver by heavily restricting the search space or make it even harder (compared to a problem without them) to converge because those need to be satisfied **exactly**. 

What is the general intuition here?

**Comment by stephane-caron** [comment]
The general intuition is: "it depends" :wink: 

It depends on both the problem and the solver. (Testing this kind of things was actually a motivation for starting qpsolvers in the first place.) For instance, active set solvers will not be slowed down by precision requirements, but they are less suited to large sparse problems. Interior point or augmented Lagrangian solvers may be affected by precision requirements on equality constraints, but some of them will carry out variable elimination internally using heuristics, and in that case they may end up solving faster with than without equality constraints. This is especially something to look out for on large sparse problems.

So, there is no general rule that I know of. If your problems have some features, you can look at [open source benchmarks](https://github.com/qpsolvers/qpbenchmark?tab=readme-ov-file#test-sets) to see if some test sets are close to your use case. (You can even submit your own problems to the benchmarks!)

**Reply by ogencoglu** [reply]
Thanks for the detailed reply!
